quota/etcd/storage/quota_storage_test.go

```diff
@@ -393,24 +393,32 @@ func TestQuotaStorage_UpdateConfigsErrors(t *testing.T) {
 
 	want := &storagepb.Configs{} // default cfgs is empty
 	if _, err := qs.UpdateConfigs(ctx, true /* reset */, updater(want)); err != nil {
-		t.Fatalf("UpdateConfigs() returned err = %v", err)
+		t.Fatalf("UpdateConfigs()= _,%v want: _,nil", err)
 	}
 
 	for _, test := range tests {
-		if _, err := qs.UpdateConfigs(ctx, false /* reset */, test.update); !strings.Contains(err.Error(), test.wantErr) {
-			// Fatal because the config has been changed, which will break all following tests.
-			t.Fatalf("%v: UpdateConfigs() returned err = %v, want substring %q", test.desc, err, test.wantErr)
-		}
+		t.Run(test.desc, func(t *testing.T) {
+			_, err := qs.UpdateConfigs(ctx, false /* reset */, test.update)
+			// First check that an error did / did not occur as the test expects.
+			if (err != nil) != (test.wantErr != "") {
+				t.Fatalf("UpdateConfigs()=_, %v, want: _, %v", err, test.wantErr)
+			}
+			// Then if we got an error check that it was the expected one.
+			if err != nil && !strings.Contains(err.Error(), test.wantErr) {
+				// Fatal because the config has been changed, which will break all following tests.
```

Comment:
This comment suggests that all subsequent test cases should be aborted, but this no longer happens since changing this to use t.Run().

Reply:
Yes, but it seemed acceptable as it's unlikely that we'll get a bug that mutates the state and returns a different error and it fails safe if we do, just with more noise.

Reply:
Could you update the comment then, since it's misleading now. Removing the comment would be fine.

I wonder why it shares the `QuotaStorage` between tests? Seems to unnecessarily expose test cases to breaking each other.

Reply:
Done. 

I suspect it used the same structure as the non-error test, where it accumulates mutations into the storage. Makes sense there but doesn't seem to be any reason do it here so I've rearranged it so the tests are independent.

Reply:
Oh I see why - there's a TestMain() that starts an embedded etcd server that is shared by all the tests, so any test case can affect any other.

Reply:
In this test there should be no changes to quota state as all the updates are expected to fail. So if it passes it should leave it untouched.